pkg/solidity-utils/contracts/test/BasePoolMathMock.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import "../math/FixedPoint.sol";
+import "../math/WeightedMath.sol";
+import "../math/BasePoolMath.sol";
+
+contract BasePoolMathMock {
+    using FixedPoint for uint256;
+
+    // It's the UniV2 invariant formula
+    // solhint-disable-next-line
+    function computeInvariantMock(uint256[] memory balancesLiveScaled18) public view returns (uint256 invariant) {
+        require(balancesLiveScaled18.length == 2, "BasePoolMathMock: INVALID_BALANCES_LENGTH");
+
+        // expected to work with 2 tokens only
+        invariant = FixedPoint.ONE;
+        for (uint256 i = 0; i < balancesLiveScaled18.length; ++i) {
+            invariant = invariant.mulDown(balancesLiveScaled18[i]);
+        }
+        // scale the invariant to 1e18
+        invariant = _sqrt(invariant) * 1e9;
+    }
+
+    // https://ethereum.stackexchange.com/questions/2910/can-i-square-root-in-solidity
+    // Babylonian Method
+    function _sqrt(uint x) internal pure returns (uint y) {
+        uint z = (x + 1) / 2;
+        y = x;
+        while (z < y) {
+            y = z;
+            z = (x / z + z) / 2;
+        }
+    }
+
+    // https://docs-v3.balancer.fi/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.html#build-your-custom-amm
+    function computeBalanceMock(
```

Comment:
This also only works with 2 tokens, since it calls `computeInvariantMock`. Consider a comment here, too